core/services/ocr2/plugins/ocr2vrf/coordinator/block_cache.go

```diff
@@ -0,0 +1,121 @@
+package coordinator
+
+import (
+	"errors"
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type blockCache[T any] struct {
+	blockEvictionWindow int64
+	cacheMu             sync.RWMutex
```

Comment:
I think you can just do with `sync.Mutex` since you only ever call `Lock` and not `RLock` (locking for reading).